### PR TITLE
Make Caption Settings Persistent

### PR DESCRIPTION
**Pull request by SebTota**
---
Made caption settings persistent.
---

**Pull Request Type**
- [x] Feature Implementation

**Related issue**
#989 

**Description**
Make the caption settings persistent across new videos and closing/re-opening the application by:
* Saving caption settings to db when the caption settings modal is closed
* Retrieving the saved caption settings each time a video is opened

**Testing**
1. Open a video with captions
2. Modify caption settings
3. Open a new video with captions and turn captions on, notice caption settings persist
4. Close the application and reopen a video, notice caption settings persist 

**Desktop (please complete the following information):**
 - OS: Mac OS
 - OS Version: 10.15.7
 - FreeTube version: Latest Development Branch

**Additional context**
At the time of creating this PR, issue #1268 may make testing difficult. Please check the comment I left on that issue if caption settings do not appear in video player.

### CODE REVIEW

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 1]
```diff
@@ -138,6 +138,10 @@ export default Vue.extend({
       return parseInt(this.$store.getters.getDefaultQuality)
     },
 
+    defaultCaptionSettings: function () {
+      return JSON.parse(this.$store.getters.getDefaultCaptionSettings)
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:142]: Any fallback logic when data corrupted for some reason?

**Review by PikachuEXE — COMMENTED** [review]
Also will saved caption settings not working/applicable for different videos?

**Comment by GilgusMaximus** [comment]
Sorry to say that, but this is not working. I can set the settings and then go to the next video and have them reset again.
I also built the app to test this, but it does not work in dev and in built version.

I do not get a setting in the settings.db file and it seems like there is also a problem within the same session now.

Additionally there have been 2 merge conflicts introduced by other commits, which are easy to fix however, as you just have to remove the merge conflict info and can leave all changes in if I am seeing this correctly.

**Comment by SebTota** [comment]
Do you mind taking a look at this [video](https://flic.kr/p/2m3Lnji) (hosted on Flickr because it's too large for GitHub). Do you not have the same results?

**Comment by GilgusMaximus** [comment]
This is how it looks for me in dev mode as well as in production mode. I also pulled the fresh branch just to make sure. From history we know that electron on Windows isn't always great with updating settings and cookies in production mode, but it usually works in dev mode.

Maybe someone else can chip in and check this out? Just to make sure this is not a problem with my OS. @Svallinn @PikachuEXE ?

https://user-images.githubusercontent.com/34301369/120933369-23d1ae80-c6fa-11eb-89ec-62a3ba67f57b.mp4

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 2]
```diff
@@ -304,6 +314,12 @@ export default Vue.extend({
             this.powerSaveBlocker = null
           }
         })
+
+        const _this = this
+        this.player.textTrackSettings.on('modalclose', function () {
+          const settings = this.getValues()
+          _this.updateDefaultCaptionSettings(JSON.stringify(settings))
+        })
```
Comment (Svallinn) [inline src/renderer/components/ft-video-player/ft-video-player.js:322]: I'd prefer if you did this instead:
```suggestion
        this.player.textTrackSettings.on('modalclose', (_) => {
          const settings = this.player.textTrackSettings.getValues()
          this.updateDefaultCaptionSettings(JSON.stringify(settings))
        })
        // OR
        this.player.textTrackSettings.on('modalclose', (event) => {
          const settings = event.target.getValues()
          this.updateDefaultCaptionSettings(JSON.stringify(settings))
        })
```

Comment (Svallinn) [inline src/renderer/components/ft-video-player/ft-video-player.js:322]: I'd prefer it if you did this instead:
```suggestion
        this.player.textTrackSettings.on('modalclose', (_) => {
          const settings = this.player.textTrackSettings.getValues()
          this.updateDefaultCaptionSettings(JSON.stringify(settings))
        })
        // OR
        this.player.textTrackSettings.on('modalclose', (event) => {
          const settings = event.target.getValues()
          this.updateDefaultCaptionSettings(JSON.stringify(settings))
        })
```

Comment (SebTota) [reply src/renderer/components/ft-video-player/ft-video-player.js]: Thank you for the suggestion! It has been fixed in the most recent commit.

**Comment by Svallinn** [comment]
LGTM. Thank you for your contribution.

**Pull request merged**